libraries/search/lib/metaheuristics/evolutionary/MOSAFamily.ts

```diff
@@ -59,7 +59,9 @@ export class MOSAFamily<T extends Encoding> extends EvolutionaryAlgorithm<T> {
       this._objectiveManager.getCurrentObjectives().size === 0 &&
       this._objectiveManager.getUncoveredObjectives().size > 0
     )
-      throw new Error(shouldNeverHappen("Objective Manager"));
+      throw new InvalidObjectStateError(
```

Comment:
It would be better to make these errors more specific (e.g., InvalidSearchAlgorithmStateError)